sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -114,10 +147,41 @@ export class BatchingReceiver extends MessageReceiver {
         reject(error);
       };
 
-      // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
-      const finalAction = (this._finalActionHandler = (): void => {
-        // clear finalActionHandler so that it can't be called multiple times.
-        this._finalActionHandler = undefined;
+      this._connectionErrorHandler = (error: AmqpError | Error): void => {
+        if (totalWaitTimer) {
```

Comment:
It'd be nice if we didn't have to duplicate this across finalAction and this handler. Looks like the only difference is that onReceiveDrain doesn't get removed in finalAction?

Reply:
My initial instinct was to move to a helper too, but yes its the diff with `onReceiveDrain` that made me pause. It does get removed in `finalAction`, but only if it is determined that drain is not needed.